src/pymor/reductors/neural_network.py

```diff
@@ -106,6 +107,70 @@ def data_to_tensor(data):
             return StationaryNeuralNetworkModel(self.neural_network, error_estimator=None, **projected_operators)
 
 
+    class StationaryNeuralNetworkOutputReductor(BasicObject):
+
+        # TODO: allow for transformation/back_transformation of outputs (and inputs?)!
+        def __init__(self, fom, training_data, validation_data,
+                     hidden_layers='[P*3, P*3]', ann_mse=None, max_restarts=10, torch_seed=None, ann_train_params=None):
+
+            assert fom.parameters and len(np.sum(fom.parameters.values())) > 0
+
+            # check that training and validation data is a tuple (mus, outputs) of a list/tuple
+            # of mus and a VectorArray of outputs, where outputs[i] == fom.output(mu[i]) is assumed
+            for data in training_data, validation_data:
+                assert isinstance(data, (list, tuple)) and len(data) == 2
+                assert isinstance(data[0], (list, tuple))
+                assert isinstance(data[1], VectorArray)
+                assert len(data[0]) == len(data[1])
+                assert all(isinstance(mu, Mu) for mu in data[0])
+                assert data[1].space == fom.output_space
+
+            assert isinstance(ann_mse, Number) or not ann_mse
+
+            # if applicable, set a common seed for the PyTorch initialization
+            # of weights and biases and further PyTorch methods for all training runs ...
+            ann_train_params = ann_train_params or {}
+            assert isinstance(ann_train_params, dict)
+            if not torch_seed and 'seed' in ann_train_params:
+                torch_seed = ann_train_params['seed']
+            ann_train_params.pop('seed', None) # has to be removed, otherwise each training uses the same
+            if torch_seed:
+                torch.manual_seed(torch_seed)
+
+            # determine the numbers of neurons in the hidden layers
+            if isinstance(hidden_layers, str):
+                hidden_layers = eval(hidden_layers, {'P': fom.parameters.dim})
```

Comment:
And then we adjust the `eval` call accordingly:
```suggestion
                hidden_layers = eval(hidden_layers, {'P': fom.parameters.dim, 'O': fom.output_space.dim})
```

Reply:
Actually, I have the same change locally by now! :)

However, training for the output seems not to work too well in my example, even when using 4 hidden layers with 100 neurons each (rescaling output did not help either), but this is a different story.

Reply:
Included suggestions.